precompiles/utils/src/lib.rs

```diff
@@ -183,8 +183,24 @@ where
 		// computations.
 		let used_weight = call
 			.dispatch(origin)
-			.map_err(|_| error("dispatched call failed"))?
+			.map_err(|original| {
+				println!("{:?}", original);
+				error("dispatched call failed")
+			})?
 			.actual_weight;
+		
+		// @librelois @nanocryk, how can I actually forward the error message here?
+		// I tried the following match statementa and a similar thing using map_err
+		// but I always get `error[E0716]: temporary value dropped while borrowed`
+		// let used_weight: Option<u64> = match call.dispatch(origin) {
+		// 	Err(e) => {
+		// 		return Err(error(&sp_std::fmt::format(format_args!(
+		// 			"Dispatched call failed: {:?}",
+		// 			e
+		// 		))));
+		// 	}
+		// 	Ok(post_info) => post_info.actual_weight,
+		// };
```

Comment:
@librelois Can you help me with this?

Reply:
Here you return a reference on a data whose owner is the scope of the return parenthesis. So it can't work, in order to return a reference, the owner of the data must survive the returning function.

This is because the error function limits the possible cases by requiring a &'static str, not allowing to return a dynamically constructed error. So here we have to make the error function more flexible by replacing its signature with :

```
pub fn error<T: Into<Cow<'static, str>>(text: T) -> ExitError
```

Reply:
Thanks for taking care of this :+1: